### DISCUSSION TITLE
Excuse me, what is the relationship between PortableExecutableReference and assembly?

### DISCUSSION THREAD

**Discussion by NMSAzulX** [dotnet/roslyn#58071] [Q&A] [answered]
Assemblies in the default domain.

> var count = AppDomain.CurrentDomain.GetAssemblies().Lenght; 
 //count  : 25

Files that need to be referenced in the actual dynamic compilation environment.

> IEnumerable<string> paths = DependencyContext .Default.CompileLibraries.SelectMany(cl => cl.ResolveReferencePaths()); 
> var count = path.Count();    
//count : 157

Is there a way to get a reference through Assembly? (Assembly.Location is null)
Now it is very troublesome to manage PortableExecutableReference  for different domains(ALC).

**Comment by svick** [comment, marked as answer]
There is [`MetadataReference.CreateFromAssembly`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.metadatareference.createfromassembly?WT.mc_id=DOP-MVP-5002735), which could look promising, but it's obsolete and doesn't work for assemblies without `Location` anyway. I have a PR open to change that: https://github.com/dotnet/roslyn/pull/57910.

I'm not aware of anything you could do in the meantime.

**Reply by NMSAzulX** [reply]
Thank you very much. I will prepare enough UT tests and wait for the new version.

**Reply by NMSAzulX** [reply]
I think it is necessary to enhance it, once the program is running, the focus should be on the assembly. All operations revolve around assemblies, such as plug-in loading, dynamic compilation, and so on. This will be very convenient for developers to use.
